### PR TITLE
make:entity Use the namespace instead of the full class name for MappingDriver

### LINKED ISSUE #1053: make:entity chooses the incorrect MappingDriver under certain conditions
When running make:entity with a new Entity the incorrect MappingDriver (Annotations / Attributes) can be selected under certain conditions.

From my tests the incorrect driver is selected when 
* The project uses attributes 
* A loaded bundle uses annotations
* The loaded bundle's namespace starts with at least character similar to `App`

See https://github.com/michaelphillpotts/mapping-driver-bug-demo as a proof of concept of the bug.

In the example there is an Bundle with the namespace **ABC** that makes use of annotations for its entities (the app uses attributes)

**To replicate:**
> symfony console make:entity FailedEntity

**Expect:** Entity to be created without error and with attributes

**Actual:**
Entity created with error 'Only annotation or attribute mapping is supported by make:entity'
Entity created with annotations

**To confirm normal behavior:**
Comment out ABCBundle in bundles.php

New entities are now created without error and with _attributes_

**Issue:**
The full class name https://github.com/symfony/maker-bundle/blob/0b0e9288a6136120d26c50904acfe235da6b9ae9/src/Doctrine/DoctrineHelper.php#L107 is passed to `getMappingDriverForNamespace` which uses `substr_compare` to find the closest match (namespace to driver) but I think this does not find the closest match if the class name is included in the comparison (and the above conditions are met regarding extra Bundles)

I suggest removing the name from full class name and just pass the namespace when calling the `getMappingDriverForNamespace` method. This also makes sense as the method names itself mentions that it returns the driver for the Namespace and not a Class

### PR DESCRIPTION
**Pull request by michaelphillpotts**
This fixes #1053

### CODE REVIEW

**Review by jrushlow — CHANGES_REQUESTED** [review]
Awesome! Could you rebase this PR please. We've fixed a number of broken tests since you've opened the PR. Thanks in advanced.

**Comment by michaelphillpotts** [comment]
Rebased the PR

File: src/Doctrine/DoctrineHelper.php [hunk 1]
```diff
@@ -104,7 +104,13 @@ public function doesClassUseDriver(string $className, string $driverClass): bool
             // if the class should be generated with attributes or annotations. If this exception is thrown, we will check based on the
             // namespaces for the given $className and compare it with the doctrine configuration to get the correct MappingDriver.
 
-            return $this->isInstanceOf($this->getMappingDriverForNamespace($className), $driverClass);
+            $classNameComponents = explode('\\', $className);
+            if (1 < \count($classNameComponents)) {
+                array_pop($classNameComponents);
+            }
+            $classNamespace = implode('\\', $classNameComponents);
```
Comment (weaverryan) [inline src/Doctrine/DoctrineHelper.php:111]: Could you add a comment about which situation this is addressing? A test would be ideal, though I don't know how easy that would be to add.

Thanks!

**Comment by michaelphillpotts** [comment]
Hi

I have added a comment to the new logic.

Regarding the tests I have had a try, but not sure of where they should be placed (I was using `EntityRegeneratorTest`), the correctness of them, and how useful they are?
The test data set `mixed_config_similar_namespace` fails on current branch and passes with this pull request.

The 2 functions I came up with are:
```php
   /**
     * @dataProvider getMappingDriverTests
     */
    public function testMappingDriverSelection(string $driverClass, ?array $annotatedPrefixes, bool $doesUseMappingDriver): void
    {
        $compatUtil = $this->createMock(PhpCompatUtil::class);

        $registry = $this->createMock(ManagerRegistry::class);
        $registry->expects($this->any())
            ->method('getManagerForClass')
            ->willThrowException(new \ReflectionException());

        $helper = new DoctrineHelper('', $compatUtil, $registry, true, $annotatedPrefixes);

        $this->assertSame($doesUseMappingDriver, $helper->doesClassUseDriver('App\\Entity\\UserProfile', $driverClass),
            sprintf('DoctrineHelper confirms that App\\Entity\\UserProfile %s use the %s driver (based on provided config)', $doesUseMappingDriver ? 'does' : 'does not', $driverClass)
        );

    }

    public function getMappingDriverTests(): \Generator
    {
        yield 'empty_config' => [
            'Doctrine\\ORM\\Mapping\\Driver\\AttributeDriver',
            null,
            false,
        ];

        $hasAttributeDriver = class_exists(\Doctrine\ORM\Mapping\Driver\AttributeDriver::class);
        $hasAnnotationDriver = class_exists(\Doctrine\ORM\Mapping\Driver\AnnotationDriver::class);

        if ($hasAttributeDriver) {
            $attributeDriver = $this->createMock(\Doctrine\ORM\Mapping\Driver\AttributeDriver::class);
            yield 'basic_config_attributes' => [
                \Doctrine\ORM\Mapping\Driver\AttributeDriver::class,
                [
                    'default' => [
                        [
                            'App\\Entity',
                            $attributeDriver
                        ]
                    ]
                ],
                true,
            ];
        }

        if ($hasAnnotationDriver) {
            $annotationDriver = $this->createMock(\Doctrine\ORM\Mapping\Driver\AnnotationDriver::class);
            yield 'basic_config_annotations' => [
                \Doctrine\ORM\Mapping\Driver\AnnotationDriver::class,
                [
                    'default' => [
                        [
                            'App\\Entity',
                            $annotationDriver
                        ]
                    ]
                ],
                true,
            ];
        }

        if ($hasAttributeDriver && $hasAnnotationDriver) {
            $attributeDriver = $this->createMock(\Doctrine\ORM\Mapping\Driver\AttributeDriver::class);
            $annotationDriver = $this->createMock(\Doctrine\ORM\Mapping\Driver\AnnotationDriver::class);

            // 2 namespaces - namespaces start with different characters (A / B)
            // New class App\Entity\Profile should be using the AttributeDriver as it belongs to App\Entity namespace
            yield 'mixed_config_different_namespaces' => [
                \Doctrine\ORM\Mapping\Driver\AttributeDriver::class,
                [
                    'default' => [
                        [
                            'App\\Entity',
                            $attributeDriver
                        ],
                        [
                            'BetaCorp\Bundle\BetaBundle\Entity',
                            $annotationDriver
                        ]
                    ]
                ],
                true,
            ];

            // 2 namespaces - namespaces start with same characters (A / A)
            // New class App\Entity\Profile should be using the AttributeDriver as it belongs to App\Entity namespace
            yield 'mixed_config_similar_namespace' => [
                \Doctrine\ORM\Mapping\Driver\AttributeDriver::class,
                [
                    'default' => [
                        [
                            'App\\Entity',
                            $attributeDriver
                        ],
                        [
                            'ABC\Bundle\ABCBundle\Entity',
                            $annotationDriver
                        ]
                    ]
                ],
                true,
            ];
        }

    }

```

**Review by jrushlow — APPROVED** [review]
Thank you @michaelphillpotts!

**Comment by UBERPHP** [comment]
I use PHP 8.1 and recently just got an error: "Class "Doctrine\Persistence\Mapping\Driver\AnnotationDriver" does not exist"
I guess this is related.
please see my reproduction:

```
#composer create-project symfony/website-skeleton:"^4.4" my_project_directory44
Creating a "symfony/website-skeleton:^4.4" project at "./my_project_directory44"
symfony/flex contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
Do you trust "symfony/flex" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?] y
Installing symfony/website-skeleton (v4.4.99)
  - Downloading symfony/website-skeleton (v4.4.99)
  - Installing symfony/website-skeleton (v4.4.99): Extracting archive
Created project in /home/vagrant/domains/hpcismart.com.local/build-archive/0/my_project_directory44
Loading composer repositories with package information
Updating dependencies
Lock file operations: 1 install, 0 updates, 0 removals
  - Locking symfony/flex (v1.18.7)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing symfony/flex (v1.18.7): Extracting archive
Generating autoload files
1 package you are using is looking for funding.
Use the `composer fund` command to find out more!

Run composer recipes at any time to see the status of your Symfony recipes.

Loading composer repositories with package information
Restricting packages listed in "symfony/symfony" to "4.4.*"
Updating dependencies
Lock file operations: 131 installs, 0 updates, 0 removals
  - Locking doctrine/annotations (1.13.2)
  - Locking doctrine/cache (2.2.0)
  - Locking doctrine/collections (1.6.8)
  - Locking doctrine/common (3.3.0)
  - Locking doctrine/dbal (3.3.6)
  - Locking doctrine/deprecations (v1.0.0)
  - Locking doctrine/doctrine-bundle (2.6.3)
  - Locking doctrine/doctrine-migrations-bundle (3.2.2)
  - Locking doctrine/event-manager (1.1.1)
  - Locking doctrine/inflector (2.0.4)
  - Locking doctrine/instantiator (1.4.1)
  - Locking doctrine/lexer (1.2.3)
  - Locking doctrine/migrations (3.5.1)
  - Locking doctrine/orm (2.12.2)
  - Locking doctrine/persistence (3.0.2)
  - Locking doctrine/sql-formatter (1.1.2)
  - Locking egulias/email-validator (3.1.2)
  - Locking friendsofphp/proxy-manager-lts (v1.0.12)
  - Locking laminas/laminas-code (4.5.1)
  - Locking monolog/monolog (1.27.0)
  - Locking myclabs/deep-copy (1.11.0)
  - Locking nikic/php-parser (v4.13.2)
  - Locking phar-io/manifest (2.0.3)
  - Locking phar-io/version (3.2.1)
  - Locking phpdocumentor/reflection-common (2.2.0)
  - Locking phpdocumentor/reflection-docblock (5.3.0)
  - Locking phpdocumentor/type-resolver (1.6.1)
  - Locking phpspec/prophecy (v1.15.0)
  - Locking phpunit/php-code-coverage (9.2.15)
  - Locking phpunit/php-file-iterator (3.0.6)
  - Locking phpunit/php-invoker (3.1.1)
  - Locking phpunit/php-text-template (2.0.4)
  - Locking phpunit/php-timer (5.0.3)
  - Locking phpunit/phpunit (9.5.20)
  - Locking psr/cache (2.0.0)
  - Locking psr/container (1.1.2)
  - Locking psr/link (1.1.1)
  - Locking psr/log (1.1.4)
  - Locking sebastian/cli-parser (1.0.1)
  - Locking sebastian/code-unit (1.0.8)
  - Locking sebastian/code-unit-reverse-lookup (2.0.3)
  - Locking sebastian/comparator (4.0.6)
  - Locking sebastian/complexity (2.0.2)
  - Locking sebastian/diff (4.0.4)
  - Locking sebastian/environment (5.1.4)
  - Locking sebastian/exporter (4.0.4)
  - Locking sebastian/global-state (5.0.5)
  - Locking sebastian/lines-of-code (1.0.3)
  - Locking sebastian/object-enumerator (4.0.4)
  - Locking sebastian/object-reflector (2.0.4)
  - Locking sebastian/recursion-context (4.0.4)
  - Locking sebastian/resource-operations (3.0.3)
  - Locking sebastian/type (3.0.0)
  - Locking sebastian/version (3.0.2)
  - Locking sensio/framework-extra-bundle (v5.6.1)
  - Locking symfony/asset (v4.4.40)
  - Locking symfony/browser-kit (v4.4.37)
  - Locking symfony/cache (v4.4.41)
  - Locking symfony/cache-contracts (v2.5.1)
  - Locking symfony/config (v4.4.41)
  - Locking symfony/console (v4.4.41)
  - Locking symfony/css-selector (v4.4.37)
  - Locking symfony/debug (v4.4.41)
  - Locking symfony/debug-bundle (v4.4.37)
  - Locking symfony/debug-pack (v1.0.10)
  - Locking symfony/dependency-injection (v4.4.41)
  - Locking symfony/deprecation-contracts (v3.0.1)
  - Locking symfony/doctrine-bridge (v4.4.41)
  - Locking symfony/dom-crawler (v4.4.39)
  - Locking symfony/dotenv (v4.4.37)
  - Locking symfony/error-handler (v4.4.41)
  - Locking symfony/event-dispatcher (v4.4.37)
  - Locking symfony/event-dispatcher-contracts (v1.1.12)
  - Locking symfony/expression-language (v4.4.41)
  - Locking symfony/filesystem (v4.4.39)
  - Locking symfony/finder (v4.4.41)
  - Locking symfony/form (v4.4.41)
  - Locking symfony/framework-bundle (v4.4.41)
  - Locking symfony/http-client (v4.4.41)
  - Locking symfony/http-client-contracts (v2.5.1)
  - Locking symfony/http-foundation (v4.4.41)
  - Locking symfony/http-kernel (v4.4.41)
  - Locking symfony/inflector (v4.4.37)
  - Locking symfony/intl (v4.4.41)
  - Locking symfony/mailer (v4.4.41)
  - Locking symfony/maker-bundle (v1.39.0)
  - Locking symfony/mime (v4.4.41)
  - Locking symfony/monolog-bridge (v4.4.37)
  - Locking symfony/monolog-bundle (v3.8.0)
  - Locking symfony/options-resolver (v4.4.37)
  - Locking symfony/orm-pack (v2.2.0)
  - Locking symfony/phpunit-bridge (v6.0.8)
  - Locking symfony/polyfill-intl-icu (v1.25.0)
  - Locking symfony/polyfill-intl-idn (v1.25.0)
  - Locking symfony/polyfill-intl-normalizer (v1.25.0)
  - Locking symfony/polyfill-mbstring (v1.25.0)
  - Locking symfony/polyfill-php72 (v1.25.0)
  - Locking symfony/polyfill-php73 (v1.25.0)
  - Locking symfony/polyfill-php80 (v1.25.0)
  - Locking symfony/polyfill-php81 (v1.25.0)
  - Locking symfony/process (v4.4.41)
  - Locking symfony/profiler-pack (v1.0.6)
  - Locking symfony/property-access (v4.4.41)
  - Locking symfony/property-info (v4.4.40)
  - Locking symfony/proxy-manager-bridge (v4.4.39)
  - Locking symfony/routing (v4.4.41)
  - Locking symfony/security-bundle (v4.4.41)
  - Locking symfony/security-core (v4.4.41)
  - Locking symfony/security-csrf (v4.4.37)
  - Locking symfony/security-guard (v4.4.37)
  - Locking symfony/security-http (v4.4.41)
  - Locking symfony/serializer (v4.4.41)
  - Locking symfony/serializer-pack (v1.0.4)
  - Locking symfony/service-contracts (v2.5.1)
  - Locking symfony/stopwatch (v4.4.38)
  - Locking symfony/test-pack (1.0.10)
  - Locking symfony/translation (v4.4.41)
  - Locking symfony/translation-contracts (v2.5.1)
  - Locking symfony/twig-bridge (v4.4.41)
  - Locking symfony/twig-bundle (v4.4.41)
  - Locking symfony/twig-pack (v1.0.1)
  - Locking symfony/validator (v4.4.41)
  - Locking symfony/var-dumper (v4.4.41)
  - Locking symfony/var-exporter (v4.4.41)
  - Locking symfony/web-link (v4.4.37)
  - Locking symfony/web-profiler-bundle (v4.4.41)
  - Locking symfony/yaml (v4.4.37)
  - Locking theseer/tokenizer (1.2.1)
  - Locking twig/extra-bundle (v3.4.0)
  - Locking twig/twig (v3.4.1)
  - Locking webmozart/assert (1.10.0)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 131 installs, 0 updates, 0 removals
  - Downloading doctrine/collections (1.6.8)
  - Downloading psr/cache (2.0.0)
  - Downloading doctrine/event-manager (1.1.1)
  - Downloading doctrine/common (3.3.0)
  - Downloading doctrine/lexer (1.2.3)
  - Downloading symfony/deprecation-contracts (v3.0.1)
  - Downloading psr/container (1.1.2)
  - Downloading symfony/service-contracts (v2.5.1)
  - Downloading symfony/stopwatch (v4.4.38)
  - Downloading symfony/polyfill-php80 (v1.25.0)
  - Downloading symfony/polyfill-php73 (v1.25.0)
  - Downloading symfony/polyfill-mbstring (v1.25.0)
  - Downloading psr/log (1.1.4)
  - Downloading symfony/filesystem (v4.4.39)
  - Downloading laminas/laminas-code (4.5.1)
  - Downloading doctrine/sql-formatter (1.1.2)
  - Downloading myclabs/deep-copy (1.11.0)
  - Downloading phar-io/version (3.2.1)
  - Downloading phar-io/manifest (2.0.3)
  - Downloading phpdocumentor/reflection-common (2.2.0)
  - Downloading phpdocumentor/type-resolver (1.6.1)
  - Downloading sebastian/recursion-context (4.0.4)
  - Downloading sebastian/exporter (4.0.4)
  - Downloading sebastian/diff (4.0.4)
  - Downloading sebastian/comparator (4.0.6)
  - Downloading webmozart/assert (1.10.0)
  - Downloading phpdocumentor/reflection-docblock (5.3.0)
  - Downloading doctrine/instantiator (1.4.1)
  - Downloading phpspec/prophecy (v1.15.0)
  - Downloading theseer/tokenizer (1.2.1)
  - Downloading sebastian/version (3.0.2)
  - Downloading nikic/php-parser (v4.13.2)
  - Downloading sebastian/lines-of-code (1.0.3)
  - Downloading sebastian/environment (5.1.4)
  - Downloading sebastian/complexity (2.0.2)
  - Downloading sebastian/code-unit-reverse-lookup (2.0.3)
  - Downloading phpunit/php-text-template (2.0.4)
  - Downloading phpunit/php-file-iterator (3.0.6)
  - Downloading phpunit/php-code-coverage (9.2.15)
  - Downloading phpunit/php-invoker (3.1.1)
  - Downloading phpunit/php-timer (5.0.3)
  - Downloading sebastian/cli-parser (1.0.1)
  - Downloading sebastian/code-unit (1.0.8)
  - Downloading sebastian/object-reflector (2.0.4)
  - Downloading sebastian/global-state (5.0.5)
  - Downloading sebastian/object-enumerator (4.0.4)
  - Downloading sebastian/resource-operations (3.0.3)
  - Downloading sebastian/type (3.0.0)
  - Downloading symfony/polyfill-php72 (v1.25.0)
  - Downloading symfony/polyfill-intl-normalizer (v1.25.0)
  - Downloading symfony/polyfill-intl-idn (v1.25.0)
  - Downloading symfony/http-client-contracts (v2.5.1)
  - Downloading symfony/event-dispatcher-contracts (v1.1.12)
  - Downloading symfony/event-dispatcher (v4.4.37)
  - Downloading symfony/polyfill-php81 (v1.25.0)
  - Downloading symfony/cache-contracts (v2.5.1)
  - Downloading doctrine/annotations (1.13.2)
  - Downloading symfony/asset (v4.4.40)
  - Downloading symfony/translation-contracts (v2.5.1)
  - Downloading symfony/profiler-pack (v1.0.6)
  - Downloading monolog/monolog (1.27.0)
  - Downloading symfony/monolog-bridge (v4.4.37)
  - Downloading symfony/debug-bundle (v4.4.37)
  - Downloading symfony/debug-pack (v1.0.10)
  - Downloading symfony/dom-crawler (v4.4.39)
  - Downloading symfony/dotenv (v4.4.37)
  - Downloading symfony/inflector (v4.4.37)
  - Downloading symfony/options-resolver (v4.4.37)
  - Downloading symfony/polyfill-intl-icu (v1.25.0)
  - Downloading egulias/email-validator (3.1.2)
  - Downloading doctrine/inflector (2.0.4)
  - Downloading symfony/maker-bundle (v1.39.0)
  - Downloading symfony/proxy-manager-bridge (v4.4.39)
  - Downloading doctrine/doctrine-bundle (2.6.3)
  - Downloading doctrine/doctrine-migrations-bundle (3.2.2)
  - Downloading symfony/orm-pack (v2.2.0)
  - Downloading symfony/security-guard (v4.4.37)
  - Downloading symfony/security-csrf (v4.4.37)
  - Downloading symfony/property-info (v4.4.40)
  - Downloading symfony/serializer-pack (v1.0.4)
  - Downloading symfony/phpunit-bridge (v6.0.8)
  - Downloading symfony/css-selector (v4.4.37)
  - Downloading symfony/browser-kit (v4.4.37)
  - Downloading phpunit/phpunit (9.5.20)
  - Downloading symfony/test-pack (1.0.10)
  - Downloading symfony/twig-pack (v1.0.1)
  - Downloading psr/link (1.1.1)
  - Downloading symfony/web-link (v4.4.37)
  - Downloading symfony/yaml (v4.4.37)
  - Installing doctrine/cache (2.2.0): Extracting archive
  - Installing doctrine/collections (1.6.8): Extracting archive
  - Installing psr/cache (2.0.0): Extracting archive
  - Installing doctrine/event-manager (1.1.1): Extracting archive
  - Installing doctrine/persistence (3.0.2): Extracting archive
  - Installing doctrine/common (3.3.0): Extracting archive
  - Installing doctrine/lexer (1.2.3): Extracting archive
  - Installing symfony/deprecation-contracts (v3.0.1): Extracting archive
  - Installing psr/container (1.1.2): Extracting archive
  - Installing symfony/service-contracts (v2.5.1): Extracting archive
  - Installing symfony/stopwatch (v4.4.38): Extracting archive
  - Installing symfony/polyfill-php80 (v1.25.0): Extracting archive
  - Installing symfony/polyfill-php73 (v1.25.0): Extracting archive
  - Installing symfony/polyfill-mbstring (v1.25.0): Extracting archive
  - Installing symfony/console (v4.4.41): Extracting archive
  - Installing psr/log (1.1.4): Extracting archive
  - Installing symfony/filesystem (v4.4.39): Extracting archive
  - Installing laminas/laminas-code (4.5.1): Extracting archive
  - Installing friendsofphp/proxy-manager-lts (v1.0.12): Extracting archive
  - Installing doctrine/deprecations (v1.0.0): Extracting archive
  - Installing doctrine/dbal (3.3.6): Extracting archive
  - Installing doctrine/migrations (3.5.1): Extracting archive
  - Installing doctrine/sql-formatter (1.1.2): Extracting archive
  - Installing myclabs/deep-copy (1.11.0): Extracting archive
  - Installing phar-io/version (3.2.1): Extracting archive
  - Installing phar-io/manifest (2.0.3): Extracting archive
  - Installing phpdocumentor/reflection-common (2.2.0): Extracting archive
  - Installing phpdocumentor/type-resolver (1.6.1): Extracting archive
  - Installing sebastian/recursion-context (4.0.4): Extracting archive
  - Installing sebastian/exporter (4.0.4): Extracting archive
  - Installing sebastian/diff (4.0.4): Extracting archive
  - Installing sebastian/comparator (4.0.6): Extracting archive
  - Installing webmozart/assert (1.10.0): Extracting archive
  - Installing phpdocumentor/reflection-docblock (5.3.0): Extracting archive
  - Installing doctrine/instantiator (1.4.1): Extracting archive
  - Installing phpspec/prophecy (v1.15.0): Extracting archive
  - Installing theseer/tokenizer (1.2.1): Extracting archive
  - Installing sebastian/version (3.0.2): Extracting archive
  - Installing nikic/php-parser (v4.13.2): Extracting archive
  - Installing sebastian/lines-of-code (1.0.3): Extracting archive
  - Installing sebastian/environment (5.1.4): Extracting archive
  - Installing sebastian/complexity (2.0.2): Extracting archive
  - Installing sebastian/code-unit-reverse-lookup (2.0.3): Extracting archive
  - Installing phpunit/php-text-template (2.0.4): Extracting archive
  - Installing phpunit/php-file-iterator (3.0.6): Extracting archive
  - Installing phpunit/php-code-coverage (9.2.15): Extracting archive
  - Installing phpunit/php-invoker (3.1.1): Extracting archive
  - Installing phpunit/php-timer (5.0.3): Extracting archive
  - Installing sebastian/cli-parser (1.0.1): Extracting archive
  - Installing sebastian/code-unit (1.0.8): Extracting archive
  - Installing sebastian/object-reflector (2.0.4): Extracting archive
  - Installing sebastian/global-state (5.0.5): Extracting archive
  - Installing sebastian/object-enumerator (4.0.4): Extracting archive
  - Installing sebastian/resource-operations (3.0.3): Extracting archive
  - Installing sebastian/type (3.0.0): Extracting archive
  - Installing symfony/polyfill-php72 (v1.25.0): Extracting archive
  - Installing symfony/polyfill-intl-normalizer (v1.25.0): Extracting archive
  - Installing symfony/polyfill-intl-idn (v1.25.0): Extracting archive
  - Installing symfony/mime (v4.4.41): Extracting archive
  - Installing symfony/http-foundation (v4.4.41): Extracting archive
  - Installing symfony/http-client-contracts (v2.5.1): Extracting archive
  - Installing symfony/event-dispatcher-contracts (v1.1.12): Extracting archive
  - Installing symfony/event-dispatcher (v4.4.37): Extracting archive
  - Installing symfony/var-dumper (v4.4.41): Extracting archive
  - Installing symfony/debug (v4.4.41): Extracting archive
  - Installing symfony/error-handler (v4.4.41): Extracting archive
  - Installing symfony/http-kernel (v4.4.41): Extracting archive
  - Installing symfony/routing (v4.4.41): Extracting archive
  - Installing symfony/finder (v4.4.41): Extracting archive
  - Installing symfony/dependency-injection (v4.4.41): Extracting archive
  - Installing symfony/polyfill-php81 (v1.25.0): Extracting archive
  - Installing symfony/config (v4.4.41): Extracting archive
  - Installing symfony/var-exporter (v4.4.41): Extracting archive
  - Installing symfony/cache-contracts (v2.5.1): Extracting archive
  - Installing symfony/cache (v4.4.41): Extracting archive
  - Installing symfony/framework-bundle (v4.4.41): Extracting archive
  - Installing doctrine/annotations (1.13.2): Extracting archive
  - Installing sensio/framework-extra-bundle (v5.6.1): Extracting archive
  - Installing symfony/asset (v4.4.40): Extracting archive
  - Installing twig/twig (v3.4.1): Extracting archive
  - Installing symfony/translation-contracts (v2.5.1): Extracting archive
  - Installing symfony/twig-bridge (v4.4.41): Extracting archive
  - Installing symfony/twig-bundle (v4.4.41): Extracting archive
  - Installing symfony/web-profiler-bundle (v4.4.41): Extracting archive
  - Installing symfony/profiler-pack (v1.0.6): Extracting archive
  - Installing monolog/monolog (1.27.0): Extracting archive
  - Installing symfony/monolog-bridge (v4.4.37): Extracting archive
  - Installing symfony/monolog-bundle (v3.8.0): Extracting archive
  - Installing symfony/debug-bundle (v4.4.37): Extracting archive
  - Installing symfony/debug-pack (v1.0.10): Extracting archive
  - Installing symfony/doctrine-bridge (v4.4.41): Extracting archive
  - Installing symfony/dom-crawler (v4.4.39): Extracting archive
  - Installing symfony/dotenv (v4.4.37): Extracting archive
  - Installing symfony/expression-language (v4.4.41): Extracting archive
  - Installing symfony/inflector (v4.4.37): Extracting archive
  - Installing symfony/property-access (v4.4.41): Extracting archive
  - Installing symfony/options-resolver (v4.4.37): Extracting archive
  - Installing symfony/polyfill-intl-icu (v1.25.0): Extracting archive
  - Installing symfony/intl (v4.4.41): Extracting archive
  - Installing symfony/form (v4.4.41): Extracting archive
  - Installing symfony/http-client (v4.4.41): Extracting archive
  - Installing egulias/email-validator (3.1.2): Extracting archive
  - Installing symfony/mailer (v4.4.41): Extracting archive
  - Installing doctrine/inflector (2.0.4): Extracting archive
  - Installing symfony/maker-bundle (v1.39.0): Extracting archive
  - Installing symfony/proxy-manager-bridge (v4.4.39): Extracting archive
  - Installing doctrine/orm (2.12.2): Extracting archive
  - Installing doctrine/doctrine-bundle (2.6.3): Extracting archive
  - Installing doctrine/doctrine-migrations-bundle (3.2.2): Extracting archive
  - Installing symfony/orm-pack (v2.2.0): Extracting archive
  - Installing symfony/process (v4.4.41): Extracting archive
  - Installing symfony/security-core (v4.4.41): Extracting archive
  - Installing symfony/security-http (v4.4.41): Extracting archive
  - Installing symfony/security-guard (v4.4.37): Extracting archive
  - Installing symfony/security-csrf (v4.4.37): Extracting archive
  - Installing symfony/security-bundle (v4.4.41): Extracting archive
  - Installing symfony/serializer (v4.4.41): Extracting archive
  - Installing symfony/property-info (v4.4.40): Extracting archive
  - Installing symfony/serializer-pack (v1.0.4): Extracting archive
  - Installing symfony/phpunit-bridge (v6.0.8): Extracting archive
  - Installing symfony/css-selector (v4.4.37): Extracting archive
  - Installing symfony/browser-kit (v4.4.37): Extracting archive
  - Installing phpunit/phpunit (9.5.20): Extracting archive
  - Installing symfony/test-pack (1.0.10): Extracting archive
  - Installing symfony/translation (v4.4.41): Extracting archive
  - Installing twig/extra-bundle (v3.4.0): Extracting archive
  - Installing symfony/twig-pack (v1.0.1): Extracting archive
  - Installing symfony/validator (v4.4.41): Extracting archive
  - Installing psr/link (1.1.1): Extracting archive
  - Installing symfony/web-link (v4.4.37): Extracting archive
  - Installing symfony/yaml (v4.4.37): Extracting archive
Package symfony/inflector is abandoned, you should avoid using it. Use EnglishInflector from the String component instead.
Generating autoload files
114 packages you are using are looking for funding.
Use the `composer fund` command to find out more!

Symfony operations: 20 recipes (427f8ef8619eb1ee18297c11c92d308b)
  - Configuring symfony/flex (>=1.0): From github.com/symfony/recipes:main
  - Configuring symfony/framework-bundle (>=4.4): From github.com/symfony/recipes:main
  - Configuring symfony/console (>=4.4): From github.com/symfony/recipes:main
  - Configuring symfony/routing (>=4.2): From github.com/symfony/recipes:main
  - Configuring doctrine/annotations (>=1.0): From github.com/symfony/recipes:main
  - Configuring sensio/framework-extra-bundle (>=5.2): From github.com/symfony/recipes:main
  - Configuring symfony/twig-bundle (>=4.4): From github.com/symfony/recipes:main
  - Configuring symfony/web-profiler-bundle (>=3.3): From github.com/symfony/recipes:main
  - Configuring symfony/monolog-bundle (>=3.3): From github.com/symfony/recipes:main
  - Configuring symfony/debug-bundle (>=4.1): From github.com/symfony/recipes:main
  - Configuring symfony/mailer (>=4.3): From github.com/symfony/recipes:main
  -  WARNING  symfony/mailer (>=4.3): From github.com/symfony/recipes:main
    The recipe for this package contains some Docker configuration.

    This may create/update docker-compose.yml or update Dockerfile (if it exists).

    Do you want to include Docker configuration from recipes?
    [y] Yes
    [n] No
    [p] Yes permanently, never ask again for this project
    [x] No permanently, never ask again for this project
    (defaults to y): y
  - Configuring symfony/maker-bundle (>=1.0): From github.com/symfony/recipes:main
  - Configuring doctrine/doctrine-bundle (>=2.3): From github.com/symfony/recipes:main
  - Configuring doctrine/doctrine-migrations-bundle (>=3.1): From github.com/symfony/recipes:main
  - Configuring symfony/security-bundle (>=4.4): From github.com/symfony/recipes:main
  - Configuring symfony/phpunit-bridge (>=5.3): From github.com/symfony/recipes:main
  - Configuring phpunit/phpunit (>=9.3): From github.com/symfony/recipes:main
  - Configuring symfony/translation (>=3.3): From github.com/symfony/recipes:main
  - Configuring twig/extra-bundle (>=v3.4.0): From auto-generated recipe
  - Configuring symfony/validator (>=4.3): From github.com/symfony/recipes:main
Unpacking Symfony packs
  - Unpacked symfony/orm-pack
  - Unpacked symfony/serializer-pack
  - Unpacked symfony/twig-pack
  - Unpacked symfony/debug-pack
  - Unpacked symfony/profiler-pack
  - Unpacked symfony/test-pack
Loading composer repositories with package information
Updating dependencies
Nothing to modify in lock file
Installing dependencies from lock file (including require-dev)
Package operations: 0 installs, 0 updates, 6 removals
  - Removing symfony/twig-pack (v1.0.1)
  - Removing symfony/test-pack (1.0.10)
  - Removing symfony/serializer-pack (v1.0.4)
  - Removing symfony/profiler-pack (v1.0.6)
  - Removing symfony/orm-pack (v2.2.0)
  - Removing symfony/debug-pack (v1.0.10)
Package symfony/inflector is abandoned, you should avoid using it. Use EnglishInflector from the String component instead.
Generating autoload files
108 packages you are using are looking for funding.
Use the `composer fund` command to find out more!

Run composer recipes at any time to see the status of your Symfony recipes.

Executing script cache:clear [OK]
Executing script assets:install public [OK]


 What's next?


Some files have been created and/or updated to configure your new packages.
Please review, edit and commit them: these files are yours.

 symfony/framework-bundle  instructions:

  * Run your application:
    1. Go to the project directory
    2. Create your code repository with the git init command
    3. Download the Symfony CLI at https://symfony.com/download to install a development web server

  * Read the documentation at https://symfony.com/doc

 symfony/mailer  instructions:

  * You're ready to send emails.

  * If you want to send emails via a supported email provider, install
    the corresponding bridge.
    For instance, composer require mailgun-mailer for Mailgun.

  * If you want to send emails asynchronously:

    1. Install the messenger component by running composer require messenger;
    2. Add 'Symfony\Component\Mailer\Messenger\SendEmailMessage': amqp to the
       config/packages/messenger.yaml file under framework.messenger.routing
       and replace amqp with your transport name of choice.

  * Read the documentation at https://symfony.com/doc/master/mailer.html

 doctrine/doctrine-bundle  instructions:

  * Modify your DATABASE_URL config in .env

  * Configure the driver (postgresql) and
    server_version (13) in config/packages/doctrine.yaml

 symfony/phpunit-bridge  instructions:

  * Write test cases in the tests/ folder
  * Use MakerBundle's make:test command as a shortcut!
  * Run the tests with php bin/phpunit


# cd my_project_directory44/
# php bin/console m:e MyNewEntity

 created: src/Entity/MyNewEntity.php
 created: src/Repository/MyNewEntityRepository.php

In DoctrineHelper.php line 180:

  Class "Doctrine\Persistence\Mapping\Driver\AnnotationDriver" does not exist


make:entity [-a|--api-resource] [-b|--broadcast] [--regenerate] [--overwrite] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--] <command> [<name>]


# php -v
PHP 8.1.5 (cli) (built: Apr 12 2022 16:19:58) (NTS gcc x86_64)
Copyright (c) The PHP Group
Zend Engine v4.1.5, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.5, Copyright (c), by Zend Technologies
    with Xdebug v3.1.4, Copyright (c) 2002-2022, by Derick Rethans

```

is it a bug or I'm missing something?

**Pull request merged**